sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/finishAnalysis.scala

```diff
@@ -52,6 +52,16 @@ object ReplaceExpressions extends Rule[LogicalPlan] {
   }
 }
 
+object ReplaceNonCorrelatedExists extends Rule[LogicalPlan] {
+  override def apply(plan: LogicalPlan): LogicalPlan = plan transformAllExpressions {
+    case exists: Exists
+      if !SubqueryExpression.hasInOrCorrelatedExistsSubquery(exists) =>
+      IsNotNull(
+        ScalarSubquery(
+          plan = Limit(Literal(1, IntegerType), exists.plan),
```

Comment:
nit: `Limit(Literal(1), Project(Literal(1), exists.plan))`

Reply:
> nit: `Limit(Literal(1), Project(Literal(1), exists.plan))`

Project need `NamedExpression`, need to construct one like  below ?
```
Project(Seq(Alias(Literal(1), s"col_${NamedExpression.newExprId}")()), exists.plan)
```